spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -661,9 +661,21 @@ def upload_watch_icon(app_version, upload_image)
     # Uploads an In-App-Purchase Review screenshot
     # @param app_id (AppId): The id of the app
     # @param upload_image (UploadFile): The icon to upload
-    # @return [JSON] the response
+    # @return [JSON] the screenshot data, ready to be added to an In-App-Purchase
     def upload_purchase_review_screenshot(app_id, upload_image)
-      du_client.upload_purchase_review_screenshot(app_id, upload_image, content_provider_id, sso_token_for_image)
+      data = du_client.upload_purchase_review_screenshot(app_id, upload_image, content_provider_id, sso_token_for_image)
+      {
+          "value" => {
+              "assetToken" => data["token"],
+              "sortOrder" => 0,
+              "type" => du_client.get_picture_type(upload_image),
+              "originalFileName" => upload_image.file_name,
+              "size" => data["length"],
+              "height" => data["height"],
+              "width" => data["width"],
+              "checksum" => data["md5"]
+          }
+      }
```

Comment:
This method looks like it is calling into the client to upload something but then has a side affect of returning a hash? Is it possible that we can separate out these two things? I'm not sure I understand the change and why this was extracted from `iap_detail.rb`?

Reply:
Hi @ohayon! Thanks for the review!
This code creating the hash was actually duplicated in both `iap_detail.rb` and in `tunes_client.rb`, which made iAP modification and creation behave in potentially different ways. So my primary goal was to reduce duplication and manage this in only one place. `itunes_client.rb` was the best one I could think of. 

Also since this method in `tunes_client.rb` is specific to the iAP functionality and was just relaying the `du_client.rb` functionality, I thought it could have been the perfect place to generate the hash too.

An alternative would be to keep the duplicated code and just use the new way to find the apple screen validation in both places. Or we can expose a new method in tunes_client that builds the hash in order to avoid the duplication and that leverages the old `upload_purchase_review_screenshot` method? Would that be better?